serpentTools/parsers/microxs.py

```diff
@@ -0,0 +1,271 @@
+            if '%' in tline:
+                tline = tline[:tline.index('%')]
+            if len(tline.split()) > 3:
+                values = self._str2num(tline)
+                # isotope, reaction type and isomeric state
+                reactionData = (values[0], values[1], values[2])
+                currXS[reactionData], currUnc[reactionData] = \
+                    splitItems(values[3:])
+        self.xsVal[univ] = currXS
+        self.xsUnc[univ] = currUnc
+
+    @staticmethod
+    def _str2num(tline):
+        """converts a string to a list of numbers."""
+        vals = [float(x) for x in tline.split()]
+        if len(tline.split()) < 2:
+            vals = vals[0]
+        return vals
+
+    def _precheck(self):
+        """do a quick scan to ensure this looks like mdx file."""
+        with open(self.filePath) as fid:
+            if fid is None:
+                raise IOError("Attempting to read on a closed file.\n"
+                              "Parser: {}\nFile: {}".format(self, self.filePath))
+            for tline in fid:
+                if 'NFY' in tline:
+                    return
+            raise SerpentToolsException("Fission yields values were not "
+                                        "found in {}".format(self.filePath))
+
+    def _postcheck(self):
+        """ensure the parser grabbed expected results."""
+        if not self.xsVal and not self.fluxRatio and not self.nfy:
+            raise SerpentToolsException("xs, fluxes and fission yields are all"
+                                        " empty in {}".format(self.filePath))
+
+    def getFY(self, parent, energy, daughter, flagEnergy=True):
+        """
+        Return a specific fission yield given the parent ID, neutron energy
+         and daughter ID
+
+        If the energy does not exist in the results, the fission yield
+        corresponding to the closest energy is returned
+
+        Parameters
+        ----------
+        parent: int or float
+            ID of the parent undergoing fission
+        energy: float
+            neutron energy in MeV
+        daughter: int or float
+            ID of the fission product
+        flagEnergy: boolean
+            If set to true, the function will return the fission yield
+            that matches to the closest energy given by the user
+
+        Returns
+        -------
+        Independent and cumulative fission yields: float or empty
+            IndYield and CumYield
+            if fission product exists, float values are returned,
+            otherwise empty values are returned
+
+        Raises
+        ------
+        SerpentToolsException:
+            If energy is a negative number
+            If parent or fission product are not found
+        """
+        IndYield, CumYield = 0, 0
+        if energy < 0:
+            raise SerpentToolsException("Energy entry {0} must be positive"
+                                        " in {1}".format(energy, self.filePath))
+        eneList = []
```

Comment:
Remove this line, as it's only used within the `if` block, where it is declared

Reply:
Removed.